src/ResourceManagement/Locks/ManagementLocksImpl.cs

```diff
@@ -0,0 +1,308 @@
+        public IManagementLock GetById(string id)
+        {
+            return Extensions.Synchronize(() => GetByIdAsync(id));
+        }
+
+        ///GENMHASH:E776888E46F8A3FC56D24DF4A74E5B74:FC67528805BD588653B69DE503549F4C
+        public async Task<IManagementLock> GetByIdAsync(string id, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            string resourceId = ResourceIdFromLockId(id);
+            string lockName = ResourceUtils.NameFromResourceId(id);
+            IManagementLock l;
+            ManagementLockObjectInner inner = null;
+            if (resourceId == null || lockName == null)
+            {
+                l = null;
+            }
+            else if (null == (inner = await Inner().GetByScopeAsync(resourceId, lockName)))
+            {
+                l = null;
+            }
+            else
+            {
+                l = WrapModel(inner);
+            }
+            return l;
+        }
+
+        ///GENMHASH:2B19D171A02814189E0329A91320316B:E509DF22AA2F28442E1D34ACD9FCD647
+        public async Task<IManagementLock> GetByResourceGroupAsync(string resourceGroupName, string name, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return WrapModel(await Inner().GetAtResourceGroupLevelAsync(resourceGroupName, name, cancellationToken));
+        }
+
+        #endregion
+
+        #region Delete
+
+        public override void DeleteById(string id)
+        {
+            Extensions.Synchronize(() => DeleteByIdAsync(id));
+        }
+
+        ///GENMHASH:4D33A73A344E127F784620E76B686786:DF0ACE79B69D4230630060F3C796E8A9
+        public async override Task DeleteByIdAsync(string id, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            string scope = ResourceIdFromLockId(id);
+            string lockName = ResourceUtils.NameFromResourceId(id);
+            if (scope != null && lockName != null)
+            {
+                await Inner().DeleteByScopeAsync(scope, lockName);
+            }
+        }
+
+        ///GENMHASH:782853D3A86D961975AF25BD353144CE:F42FD335BA3F56DE82610403C83DB1CD
+        public async Task<IEnumerable<string>> DeleteByIdsAsync(IList<string> ids, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (ids == null || ids.Count == 0)
+            {
+                return new List<string>();
+            }
+
+            IEnumerable<Task<string>> tasks = new List<Task<string>>();
+            List<string> ids1 = new List<string>();
+            List<Task> tts = new List<Task>();
+
+            foreach (var id in ids)
+            {
+                string lockName = ResourceUtils.NameFromResourceId(id);
+                string scopeName = ResourceIdFromLockId(id);
+                if (lockName != null && scopeName != null)
+                {
+                    Task t = Inner().DeleteByScopeAsync(scopeName, lockName).ContinueWith((Task parent) =>
```

Comment:
you should always pass following parameters to the ContinueWith methods after the lambda expression:
```
                cancellationToken,
                TaskContinuationOptions.ExecuteSynchronously,
                TaskScheduler.Default
```

Reply:
ok